app/lib/features/pins/pages/create_pin_page.dart

```diff
@@ -0,0 +1,353 @@
+            ],
+          ),
+        ),
+      ),
+      confirmActionTitle: 'Create Pin',
+      cancelActionTitle: null,
+      confirmActionOnPressed: () async {
+        if (_formKey.currentState!.validate()) {
+          _handleCreatePin();
+        }
+      },
+    );
+  }
+
+  Widget _buildTitleField() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        const Padding(
+          padding: EdgeInsets.only(bottom: 5),
+          child: Text('Title'),
+        ),
+        InputTextField(
+          hintText: 'Pin Name',
+          key: CreatePinPage.titleFieldKey,
+          textInputType: TextInputType.text,
+          controller: _titleController,
+          validator: (value) => (value != null && value.trim().isNotEmpty)
+              ? null
+              : 'Please enter a title',
+        ),
+      ],
+    );
+  }
+
+  Widget _buildAttachmentField() {
+    return InkWell(
+      onTap: () => PinUtils.showAttachmentSelection(context, ref),
+      child: Container(
+        height: 100,
+        width: double.infinity,
+        decoration: BoxDecoration(
+          color: Theme.of(context).colorScheme.primaryContainer,
+          border: Border.all(
+            color: Theme.of(context).colorScheme.primary,
+            strokeAlign: BorderSide.strokeAlignOutside,
+          ),
+          borderRadius: BorderRadius.circular(12),
+        ),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Icon(
+              Atlas.file_arrow_up_thin,
+              size: 14,
+              color: Theme.of(context).colorScheme.neutral5,
+            ),
+            Text(
+              'Upload File',
+              style: Theme.of(context)
+                  .textTheme
+                  .labelLarge!
+                  .copyWith(color: Theme.of(context).colorScheme.neutral5),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildLinkField() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        const Padding(
+          padding: EdgeInsets.only(bottom: 5),
+          child: Text('Link'),
+        ),
+        InputTextField(
+          hintText: 'https://',
+          key: CreatePinPage.urlFieldKey,
+          textInputType: TextInputType.url,
+          controller: _linkController,
+          validator: (value) =>
+              hasLinkOrText() ? null : 'Text or URL must be given',
+        ),
+      ],
+    );
+  }
+
+  Widget _buildDescriptionField() {
+    final labFeature = ref.watch(featuresProvider);
+    bool isActive(f) => labFeature.isActive(f);
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        const Padding(
+          padding: EdgeInsets.only(bottom: 5),
+          child: Text('Description'),
+        ),
+        Visibility(
+          visible: isActive(LabsFeature.pinsEditor),
```

Comment:
I think we can safely switch this to the new HtmlEditor only now ... Remove the entire labs feature around it.

Reply:
Will do in subsquent PR's!